llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -24079,11 +24080,9 @@ static bool findMoreOptimalIndexType(const MaskedGatherScatterSDNode *N,
 
 static SDValue performMaskedGatherScatterCombine(
     SDNode *N, TargetLowering::DAGCombinerInfo &DCI, SelectionDAG &DAG) {
-  MaskedGatherScatterSDNode *MGS = cast<MaskedGatherScatterSDNode>(N);
-  assert(MGS && "Can only combine gather load or scatter store nodes");
```

Comment:
We should probably keep this assertion, while changing the message to include histograms. I also agree with MacDue that we don't really need the change that moves the cast downwards.

Reply:
Oh the assertion isn't necessary as the `cast` call will do that. I do think moving the `MGS` declaration is unnecessary though.